File: app/src/main/java/com/vladislav/workoutassistant/data/models/WorkoutProgram.java
package com.vladislav.workoutassistant.data.models;

import com.vladislav.workoutassistant.data.db.entity.Set;
import com.vladislav.workoutassistant.data.db.entity.Workout;

import java.util.List;

import androidx.room.Embedded;
import androidx.room.Relation;

public class WorkoutProgram {

    @Embedded
    public Workout workoutInfo;

    @Relation(parentColumn = "id", entityColumn = "workout_id", entity = Set.class)
    private List<WorkoutSet> mWorkoutSets;

    /* GETTERS */
    public List<WorkoutSet> getWorkoutSets() {
        return mWorkoutSets;
    }

    /* SETTERS */
    public void setWorkoutSets(List<WorkoutSet> workoutSets) {
        mWorkoutSets = workoutSets;
    }
}

File: app/src/main/java/com/vladislav/workoutassistant/data/models/WorkoutProgram.kt
package com.vladislav.workoutassistant.data.models

import com.vladislav.workoutassistant.data.db.entity.Set
import com.vladislav.workoutassistant.data.db.entity.Workout

import androidx.room.Embedded
import androidx.room.Relation

class WorkoutProgram {

    @Embedded
    var workoutInfo: Workout? = null

    @Relation(parentColumn = "id", entityColumn = "workout_id", entity = Set::class)
    var workoutSets: List<WorkoutSet>? = null
}
